tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,210 @@
+					NetworkName: secondaryNetworkName,
+				},
+			},
+		}
+
+		secondaryLinuxBridgeInterface1 := kvirtv1.Interface{
+			Name: linuxBridgeNetwork1.Name,
+			InterfaceBindingMethod: kvirtv1.InterfaceBindingMethod{
+				Bridge: &kvirtv1.InterfaceBridge{},
+			},
+			MacAddress: "02:00:a0:a0:a0:a0",
+		}
+
+		secondaryLinuxBridgeInterface2 := kvirtv1.Interface{
+			Name: linuxBridgeNetwork2.Name,
+			InterfaceBindingMethod: kvirtv1.InterfaceBindingMethod{
+				Bridge: &kvirtv1.InterfaceBridge{},
+			},
+			MacAddress: "02:00:a1:a1:a1:a1",
+		}
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			nad := newNetworkAttachmentDefinition(secondaryNetworkName)
+			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			changeMacCmd := "sudo ip link set dev eth0 address " + updatedMac + "\n"
+			createDummyInterfaceCmd := "ip link add dummy0 type dummy\n" +
+				"ip link set dev dummy0 address " + dummyInterfaceMac + "\n"
+			setnsCmd := "ip netns add testns\n" +
+				"ip link set eth2 netns testns\n"
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(defaultInterfaceName)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface1),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface2),
+				libvmi.WithNetwork(&linuxBridgeNetwork1),
+				libvmi.WithNetwork(&linuxBridgeNetwork2),
+				libvmi.WithCloudInitNoCloudUserData(
+					"#!/bin/bash\n"+changeMacCmd+
+						createDummyInterfaceCmd+
+						setnsCmd,
+					false))
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        originalMac,
+					Name:       defaultInterfaceName,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryLinuxBridgeInterface1.MacAddress,
+					Name:          secondaryLinuxBridgeInterface1.Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryLinuxBridgeInterface2.MacAddress,
+					Name:       secondaryLinuxBridgeInterface2.Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           updatedMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "dummy0",
+					MAC:           dummyInterfaceMac,
+				},
+			}
```

Comment:
Can you split it in a way we can easily see what are the interfaces expected to be in sync (domain + guest) and the ones only on the domain or guest? Maybe it will not be needed if the const names are updated.

Reply:
I will start with changing the names, we can't split it, unless we reorder the vmi.status.interfaces as well
The reason is that the ones which appear on the domain are ordered as their spec order when the vmi status is built,
Where the guest agent ones order is random (see https://github.com/kubevirt/kubevirt/pull/6793#discussion_r768765169)

Another thing i might be able to do is switch between the purpose of secondaryLinuxBridgeInterface1 and secondaryLinuxBridgeInterface2 , so it will be domain, domain, domainAndGa, guest-agent, guest-agent
instead domain, domainAndGa, domain, guest-agent, guest-agent